src/sema/yul/statements.rs

```diff
@@ -108,6 +108,13 @@ pub(crate) fn resolve_yul_statement(
                 ns,
             )?;
             resolved_statements.push(resolved_switch.0);
+            ns.diagnostics.push(
+                Diagnostic::error(
+                    switch_statement.loc,
+                    "switch statements have no implementation in code generation yet. Please, file a GitHub issue \
+                    if there is urgent need for such a feature".to_string()
+                )
+            );
```

Comment:
@seanyoung Due to this error, Solang will never reach the `unimplemented!` function in `codegen/yul/statement.rs`